src/extension.ts

```diff
@@ -98,84 +97,46 @@ export function activate(context: vscode.ExtensionContext) {
       );
 
       lspClient.onReady().then(() => {
-        lspClient.onNotification('caNotification', respData => {
-          if (
-            respData &&
-            respData.hasOwnProperty('diagCount') &&
-            vscode.window.activeTextEditor &&
-            ((respData.diagCount > 0 && respData.diagCount !== diagCountInfo) ||
-              !onFileOpen ||
-              (onFileOpen &&
-                onFileOpen.indexOf(
-                  vscode.window.activeTextEditor.document.fileName
-                ) === -1))
-          ) {
-            diagCountInfo = respData.diagCount;
-            onFileOpen.push(vscode.window.activeTextEditor.document.fileName);
-            showInfoOnfileOpen(respData.data);
+        const notifiedFiles = new Set<string>();
+        const canShowPopup = (notification: CANotification): boolean => {
+          const hasAlreadyShown = notifiedFiles.has(vscode.window.activeTextEditor.document.fileName);
+          return notification.hasWarning() && !hasAlreadyShown;
+        }
+
+        const showVulnerabilityFoundPrompt = async (msg: string) => {
+          const selection = await vscode.window.showInformationMessage(`${msg}. Powered by [Snyk](${registrationURL})`, StatusMessages.FULL_STACK_PROMPT_BUTTON);
+          if (selection === StatusMessages.FULL_STACK_PROMPT_BUTTON) {
+            vscode.commands.executeCommand(Commands.TRIGGER_FULL_STACK_ANALYSIS);
           }
-          if (!caNotif) {
-            vscode.window.withProgress(
-              {
-                location: vscode.ProgressLocation.Window,
-                title: StatusMessages.EXT_TITLE
-              },
-              progress => {
-                caNotif = true;
-                progress.report({
-                  message: 'Checking for security vulnerabilities ...'
-                });
-
-                setTimeout(() => {
-                  progress.report({
-                    message: respData.data
-                  });
-                }, 1000);
-
-                let p = new Promise(resolve => {
-                  setTimeout(() => {
-                    caNotif = false;
-                    resolve();
-                  }, 1600);
-                });
-                return p;
-              }
-            );
+        };
+
+        lspClient.onNotification('caNotification', respData => {
+          const notification = new CANotification(respData);
+          caStatusBarProvider.showSummary(notification.statusText());
+          if (canShowPopup(notification)) {
+            showVulnerabilityFoundPrompt(notification.popupText());
+            // prevent further popups.
+            notifiedFiles.add(vscode.window.activeTextEditor.document.fileName);

```

Comment:
Dont we want to show the notification if the number of vulnerabilities changes.

Reply:
Notification will be shown only once per manifest. Further changes will be updated only via status bar.

Reply:
Once per manifest per session, is that correct?

Reply:
Exactly.

Reply:
@prashbnair @mathur07 , I just found a bug in this implementation. If you immediately switch to different file just after opening a manifest, the popup shown again and again.

Solution would be to use actual file uri from lsp instead of `vscode.window.activeTextEditor.document.fileName`.

Reply:
Corresponding LSP PR.